### PR TITLE
#64 Create user

### PR DESCRIPTION
**Pull request by SergioGueruiz**
:information_source:  This issue is part of the Epic https://github.com/kommitters/tickspot.js/issues/59.
Objective

Create a module to create a user from the Tickspot API.

Documentation:

- [POST/users.json](https://github.com/tick/tick-api/blob/master/sections/users.md#create-user)

CheckList

- [x] Create a method to create a user.
- [x] Create tests.

### CODE REVIEW

**Comment by einerzg** [comment]
@miguelnietoa please help me with your review here!!!

File: src/v2/resources/users.js [hunk 1]
```diff
@@ -76,6 +76,44 @@ class Users extends BaseResource {
...
+    if (!firstName) throw new Error('first name field is missing');
+    if (!lastName) throw new Error('last name field is missing');
+    if (!email) throw new Error('email field is missing');
+
+    const body = {
+      firstName,
+      lastName,
+      email,
+      ...((typeof admin === 'boolean' && { admin }) || { admin: false }),
+      ...((billableRate && { billableRate }) || { billableRate: '0.0' }),
```
Comment (miguelnietoa) [inline src/v2/resources/users.js:109]: The fields `admin` and `billable_rate` seems to be ignored by the Tick API, please review if these fields are allowed in the create-user endpoint.

**Comment by SergioGueruiz** [comment]
## Update
The fields `admin` and `billable_rate` are ignored by the API when we try to create an user, so we decide to delete them from the request.

File: src/v2/resources/users.js [hunk 2]
```diff
@@ -76,6 +76,40 @@ class Users extends BaseResource {
...
   }
+
+  /**
+   * This will create a new user.
+   * @param {object} User contains the fields to send in the body request.
+   *    [Required] firstName
+   *    [Required] lastName
+   *    [Required] email
+   *    [Optional] admin
+   *    [Optional] billableRate
```
Comment (miguelnietoa) [inline src/v2/resources/users.js:87]: Please remove `admin` and `billabeRate` from function documentation!

File: src/v2/resources/users.js [hunk 3]
```diff
@@ -76,6 +76,40 @@ class Users extends BaseResource {
...
+  }, responseCallback) {
+    if (!first_name) throw new Error('first name field is missing');
+    if (!last_name) throw new Error('last name field is missing');
+    if (!email) throw new Error('email field is missing');
+
+    const body = {
+      first_name,
+      last_name,
+      email,
+    };
```
Comment (miguelnietoa) [inline src/v2/resources/users.js:106]: We should keep the **camelCase** style in the parameters of the functions, however at the moment of building the body, the API receives the fields in **snake_case**

**Review by miguelnietoa — CHANGES_REQUESTED** [review]

**Comment by miguelnietoa** [comment]
> ## Update
> The fields `admin` and `billable_rate` are ignored by the API when we try to create an user, so we decide to delete them from the request.

Agree, I have reported an issue so let's keep alert to the response!
In the meantime, let's leave those fields outside.

Issue: https://github.com/tick/tick-api/issues/56

**Review by miguelnietoa — APPROVED** [review]
Thanks for your contribution, @SergioGueruiz! 🎉 🎉

**Comment by miguelnietoa** [comment]
@SergioGueruiz, could you please sign all commits? This is a security rule to merge pull requests 😅

**Review by miguelnietoa — APPROVED** [review]

**Pull request merged**